ui/components/app/alert-system/confirm-alert-modal/confirm-alert-modal.tsx

```diff
@@ -122,19 +122,26 @@ export function ConfirmAlertModal({
   ownerId,
 }: ConfirmAlertModalProps) {
   const t = useI18nContext();
-  const { alerts, unconfirmedDangerAlerts } = useAlerts(ownerId);
+  const { fieldAlerts, alerts, isAlertConfirmed } = useAlerts(ownerId);
 
   const [confirmCheckbox, setConfirmCheckbox] = useState<boolean>(false);
 
-  // if there are multiple alerts, show the multiple alert modal
+  const unconfirmedDangerFieldAlerts = fieldAlerts.filter(
+    (alert) =>
+      !isAlertConfirmed(alert.key) && alert.severity === Severity.Danger,
+  );
+
+  const hasBlockingAlerts = fieldAlerts.some((alert) => alert.isBlocking);
+
+  // if there are unconfirmed danger alerts, show the multiple alert modal
   const [multipleAlertModalVisible, setMultipleAlertModalVisible] =
-    useState<boolean>(unconfirmedDangerAlerts.length > 1);
+    useState<boolean>(unconfirmedDangerFieldAlerts.length > 0);
```

Comment:
Do we also want to show the multiple alert modal if there is more than one general alert, since that would mean neither was rendered in the confirmation already?

Reply:
We won't show general alerts in the multiple alert modal. @SayaGT asked me to remove the logic behind the general alert when there is more than one.
I'm still working on that and will probably do it in a separate PR cause this one already got too big with all those fixes.